content/250-postgres/1200-more/1000-faq.mdx

```diff
@@ -73,6 +75,148 @@ Yes, read and write queries are counted equally as _operations_ and are billed t
+
+### What strategies can I use to optimize cost per operation?
+
+Prisma Postgres bills by operation. The more you can perform using a single operation, the lower your bill. Some tips to reduce the number of operations:
+
+- [**Batch your writes**](/orm/prisma-client/queries/transactions#batchbulk-operations) with `createMany`, `updateMany`, or `deleteMany` instead of looping over single-row calls.
+
+  ```ts
+  // One operation, three users
+  await prisma.user.createMany({
+    data: [
+      { name: 'Alice' },
+      { name: 'Bob'   },
+      { name: 'Carol' },
+    ],
+  })
+  ```
+
+- **Use nested-relation helpers** such as [`connectOrCreate`](/orm/reference/prisma-client-reference#connectorcreate) or [`set`](/orm/reference/prisma-client-reference#set) to create or link related records in a single operation.
+
+  ```ts
+  // Post and (if needed) its author, all in one request
+  await prisma.post.create({
+    data: {
+      title: 'Hello World',
+      author: {
+        connectOrCreate: {
+          where:  { email: 'alice@example.com' },
+          create: { name: 'Alice', email: 'alice@example.com' },
+        },
+      },
+    },
+  })
+  ```
+
+- **Prefer regular (array) [transactions](/orm/prisma-client/queries/transactions#transaction-api) over [interactive transactions](/orm/prisma-client/queries/transactions#interactive-transactions)** when the individual queries don't depend on each other.
+
+  ```ts
+  // Interactive transaction: counted as 2 operations
+  await prisma.$transaction(async (tx) => {
+    await tx.user.create({ data: { name: 'Alice' } })
+    await tx.post.create({ data: { title: 'Hello', authorId: 1 } })
+  })
+
+  // Array transaction: counted as 1 operation
+  await prisma.$transaction([
+    prisma.user.create({ data: { name: 'Alice' } }),
+    prisma.post.create({ data: { title: 'Hello', authorId: 1 } }),
+  ])
+  ```
+
+If a later query needs the result of an earlier one (for example, you need the user ID you just created), stick with an interactive transaction for correctness. Otherwise, batching and array transactions let you collapse multiple queries into a single billed operation, keeping both your operation count, and your cost down.
+
+### Is there a sample workload to estimate my expected charges?
+
+We will demonstrate three example workloads and estimate bills for small, medium, and large workloads. Each combine a realistic number of monthly active users (MAUs), a typical level of daily activity per user, and a rounded storage footprint. 
+
+We will use the following equations to estimate the monthly bill:
+
+
+```
+total_ops            = MAUs x actions_per_user_per_day x 30      
+billable_ops         = MAX(total_ops - 100_000, 0)
```

Comment:
```suggestion
billable_ops         = total_ops - 100_000
```

While yes, this may go into negative, mathematically, we're demonstrating a calculation here, not writing code, so it's ok and easier to read.